calypso/api.go

```diff
@@ -87,16 +89,25 @@ func (c *Client) CreateLTS(ltsRoster *onet.Roster, darcID darc.ID, signers []dar
 	return reply, nil
 }
 
-// Authorise adds a ByzCoinID to the list of authorized IDs for each
-// server in the roster. The AuthoriseByzcoinID service refuses requests
-// that do not come from localhost.
+// Authorize adds a ByzCoinID to the list of authorized IDs in the server. To
+// be accepted, the request must be signed by the private key stored in
+// private.toml. For testing purposes, the environment variable can be set:
+//   COTHORITY_ALLOW_INSECURE_ADMIN=true
+// this disables the signature check.
 //
 // It should be called by the administrator at the beginning, before any other
 // API calls are made. A ByzCoinID that is not authorised will not be allowed to
 // call the other APIs.
-func (c *Client) Authorise(who *network.ServerIdentity, what skipchain.SkipBlockID) error {
-	reply := &AuthoriseReply{}
-	err := c.c.SendProtobuf(who, &Authorise{ByzCoinID: what}, reply)
+func (c *Client) Authorize(who *network.ServerIdentity, what skipchain.SkipBlockID) error {
+	reply := &AuthorizeReply{}
+	sig, err := schnorr.Sign(cothority.Suite, who.GetPrivate(), what)
```

Comment:
Yes, he could. Should I add a timestamp and only allow authorization requests that are not older than x minutes? Or do we say that an attacker would have to be able to get hold of the signature in the first place, and that this is very unlikely? Given that we have tls-enabled websocket communication, the attacker would have to be _in_ the conode or the client, and then he can do anything he wants anyway.

Reply:
I would prefer protect the signature because even if an attacker is inside and can do whatever he wants, that means he can get signatures from other conodes, even after being thrown away by the admin. A timestamp would be perfectly fine here I guess.

Reply:
An attacker cannot use a signature from another conode, as it is verified using the public key of the conode. So he can only re-authorize the LTS on the conode he controls. Which he can do anyway...

Reply:
Yes you're right. But still, I think it's not totally useless to protect this signature a bit more because if a leak appears for any reason. It's always better to have multiple layers of security and it is not too hard to change I think.

Reply:
OK, added a timestamp to the signature. I'm not totally convinced it is really needed, and just putting one more element in a signature does not necessarily make it better.